user.go

```diff
@@ -39,6 +44,58 @@ func (a *App) GetUser(userID string) (*User, error) {
 	}
 }
 
+// GetUserByIdentifier gets a user using their identifier
+// returns user on success, error on failure
+func (a *App) GetUserByIdentifier(identifier string) (*User, error) {
+	var errorText string
+	message := "failed to get Passage User By Identifier"
+	limit := 1
+	lowerIdentifier := strings.ToLower(identifier)
+	res, err := a.client.ListPaginatedUsersWithResponse(
+		context.Background(),
+		a.ID,
+		&ListPaginatedUsersParams{
+			Limit:      &limit,
+			Identifier: &lowerIdentifier,
+		},
+	)
```

Comment:
I just saw that we always transform emails to lower when we create users. This can be ignored

Reply:
I opted not to use `like` bc:
1. we want to find the exact match and not variations of it
2.  `like` is case sensitive so it still would not find the lower case version of the provided identifier. The reason why line 53 is using `toLower` is bc in the create we consistently lowercase the identifier.

Reply:
Do we have to clean or do some validation on the sdk side to make sure that the pagination prepended operations are not abused to find any user by doing something like `like: A%`?

Reply:
Our `like:` operator is case sensitive but we support a case insensitive like, `like:(?i)`. But either way since we transform emails to lowercase when creating a user your `strings.ToLower(identifier)` works just fine here.

Reply:
I did not realize that was for case insensitivity. I have updated the code